theLudovyc/Building/Building.gd

```diff
@@ -10,6 +10,8 @@ signal selected(type)
 var event_bus: EventBus
 
 var is_selected := false
+var current_ticks: int = 0
```

Comment:
Hello :wave: 

I made an architecture like MVC, with backend and frontend. The class building2d is only here for the frontend part. Please do not add logic in here :blush: 

This is why the event_bus exist also.

https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller

Reply:
Adjusted the code in order to fit the MVC pattern.